app/src/main/java/com/asfoundation/wallet/ui/gamification/LevelsAdapter.kt

```diff
@@ -59,15 +65,37 @@ class LevelsAdapter(private val context: Context,
     }
   }
 
+  /**
+   * set the content to display regarding the several levels
+   * with this one can update the view holders several times
+   */
+  fun setLevelsContent(_hiddenLevels: List<LevelModel>, _shownLevels: List<LevelModel>,
+                       _amountSpent: BigDecimal, _currentLevel: Int,
+                       _nextLevelAmount: BigDecimal?) {
+    activeLevelList.clear()
+    hiddenLevels = _hiddenLevels
+    shownLevels = _shownLevels
+    amountSpent = _amountSpent
+    currentLevel = _currentLevel
+    nextLevelAmount = _nextLevelAmount
+    if (reachedLevelsToggled) {
+      activeLevelList.addAll(hiddenLevels)
+      activeLevelList.addAll(shownLevels)
+    } else activeLevelList.addAll(shownLevels)
+    notifyDataSetChanged()
+  }
+
```

Comment:
The adapter should not have any reference other than the list it has. This must be done differently.
My suggestion (we can discuss this) is:
Change LevelModel to LevelItem (this is optional of course, but since we are doing big changes here might as well)
Create three classes that extend LevelItem (CurrentLevelItem, ReachedLevelItem, UnreachedLevelItem). This way you can also remove the viewType parameter from the Item as you can check through the instance.
Then have the CurrentLevelItem, have AmountSpent, CurrentLevel and nextLevelAmount.
Also, I'm not sure, but I think you don't need to clear and add all of them again you can just use this.list = list. Again not sure.
Also, we aren't using the "_" notation, I believe that's more for Python. I prefer using this.x = x instead.